packages/api-key/src/models/api-key.ts

```diff
@@ -1,33 +1,68 @@
-import { generateEntityId } from "@medusajs/utils"
+import {
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 
 import {
   BeforeCreate,
   Entity,
   OnInit,
   PrimaryKey,
   Property,
+  Enum,
 } from "@mikro-orm/core"
 
-// TODO:
+const TypeIndex = createPsqlIndexStatementHelper({
+  tableName: "api_key",
+  columns: "type",
+})
+
 @Entity()
 export default class ApiKey {
   @PrimaryKey({ columnType: "text" })
   id: string
 
+  @Property({ columnType: "text" })
+  token: string
```

Comment:
I introduced a token field, so we don't rely on the ID being the token as well. This worked fine for publishable keys, but naturally it doesn't work for secret and restricted keys, so it's better for all of them to rely on a single column - token.